samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -293,31 +535,19 @@ static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap
 	printk("Received BASE with %d subgroups from broadcast sink %p\n",
 	       bt_bap_base_get_subgroup_count(base), sink);
 
-#if defined(CONFIG_LIBLC3)
-	struct bt_audio_codec_cfg codec_cfg = {0};
-
-	err = bt_bap_base_foreach_subgroup(base, base_subgroup_cb, &codec_cfg);
-	if (err != 0 && err != -ECANCELED) {
-		printk("Failed to parse subgroups: %d\n", err);
-		return;
-	} else if (codec_cfg.id != BT_HCI_CODING_FORMAT_LC3) {
-		/* No subgroups with LC3 was found */
-		printk("Did not parse an LC3 codec\n");
-		return;
-	}
-
-	err = lc3_enable(&codec_cfg);
-	if (err < 0) {
-		printk("Error: cannot enable LC3 codec: %d", err);
+#if defined(CONFIG_TARGET_BROADCAST_CHANNEL) && defined(CONFIG_USE_AUTOMATIC_BROADCAST_SYNC)
+	err = base_get_first_valid_bis(base, &base_bis_index_bitfield);
```

Comment:
This function may return `0` where `base_bis_index_bitfield` gets set to 0, which isn't handled

Reply:
I dont think I see how that happens, can you elaborate?

`base_get_first_valid_bis` only returns 0 if the foreach returned `-ECANCELED`. Even if the bis index was zero, the return is a `1` bitshifted by the index - which would become ´1´ in that case.

What am I missing? :)

Reply:
As I read it, it works as following

`base_get_first_valid_bis` returns 0 if `bt_bap_base_foreach_subgroup` returns 0. 
`bt_bap_base_foreach_subgroup` returns 0 if `bt_bap_base_subgroup_foreach_bis` returns anything but `-ECANCELED` (including 0). 
`bt_bap_base_subgroup_foreach_bis` returns 0 if channel allocation is not found. 

(Please correct me if the above is not correct)
So that means that if channel allocation is not found in the BIS codec configuration, then `base_get_first_valid_bis` returns 0 and the BIS index is not set, which means that this function can return 0, and `base_bis_index_bitfield` remain at `0`. In this case, `sem_bis_sync_requested` may still be set, even though `base_bis_index_bitfield` is 0, which is not correct. 

Summary: If the channel allocation is not set in any of the BIS (which is still a valid configuration/BASE), then `base_bis_index_bitfield` ends up being `0`, and then we give the `sem_bis_sync_requested` which is incorrect.

Reply:
>base_get_first_valid_bis returns 0 if bt_bap_base_foreach_subgroup returns 0.
This is incorrect :)
`base_get_first_valid_bis` returns 0 only if `bt_bap_base_foreach_subgroup` returns `-ECANCELED`. At time of your original comment it returned ´-1´ when `bt_bap_base_foreach_subgroup` returned 0, but has since been changed to return `-ENOENT`.